Last pull request from Tests implementation

gaabs opened the pull request:
Important note: the pull requests I've received contained too many errors (even compilation ones). I got the code back to compile and run, but couldn't get Tests to work again. 
I keep getting this error: Fatal error running tests: Could not find ApplicationContext, configure Grails correctly first.

Edit: Made several changes and got tests to run again.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pauloborba/ta/14)

<!-- Reviewable:end -->

ncrneto committed:
AutoavaliacaoPrazo

Documentacao sobre definicao de prazo de autoavaliacao

dansq committed:
Merge pull request #2 from psg2/master

Merge feito
4 New Features : Manual Concept Input, Edit Auto Evaluation, Import Concepts from Spreadsheets and Registration of Teachers

diest committed:
Merge pull request #2 from ncrneto/autoavaliacaoDocumentacao

AutoavaliacaoPrazo

cjvg committed:
Merge pull request #1 from dansq/master

Pull request 7 >> 3

cjvg committed:
Merge pull request #2 from diest/master

Commits dos alunos 9, 10 e 12

diest committed:
Review AccessControl.feature

Elementos do cenário como botões não deveriam ser especificados desta maneira, pois ela isola as plataformas. É uma boa prática deixar os cenários aplicáveis a várias plataformas diferentes.

diest committed:
Review AutoavaliacaoPrazo.feature

A rigor, os cenários estão com passos longos os quais poderiam ser divididos em dois com 'And' / 'E'. Só faltaram os cenários de interface gráfica.

Commit:
AutoEvaluation tests - First changes

Commit:
AutoEvaluation tests - Second change

imvm committed:
Update Notifications.feature

imvm committed:
Synchronizing local repository

Commit:
Import Concept Step file created

Commit:
Created Import Concepts and necessary auxiliary classes. Modified the feature to include parameters.

dansq committed:
Test implementation for RegisterDiscipline.

Changes in .feature file following monitor's observations (removed ":" from Given, When, Then and And).

Created auxiliary .groovy files for the test.

dansq committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	test/cucumber/RegisterDiscipline.feature

Commit:
Passos de teste
Classes auxiliares
Adaptação dos cenários de feature

Commit:
Ajustes finais

Commit:
Remoção de ações em branco.

Commit:
Created Sheet and SheetBuilder and made small corrections to ImportConcepts and ConceptController.

renatodms committed:
Groovy de testes da Feature RegistrationOfTeachers

imvm committed:
Merge pull request #3 from renatodeyvson/master

Groovy de testes da Feature RegistrationOfTeachers

imvm committed:
Beggining notification feature test implementation

imvm committed:
Beggining notification feature test implementation

imvm committed:
Merge branch 'master' of https://github.com/imvm/TA

cjvg committed:
Merge pull request #3 from gaabs/master

Import Concepts from SpreedSheet Test + auxiliary classes

cjvg committed:
Changes for auto evaluation

cjvg committed:
Change for auto evaluation 2

cjvg committed:
Merge remote-tracking branch 'origin/master'

Commit:
Update EvaluationComparison.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update ShowComparisonPage.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInputSteps.groovy

Commit:
Update ManualInputPage.groovy

Commit:
Edited chromedriver file and gebconfig to make tests run.

Corrected a few errors on features, tests and auxiliary classes from ImportConcepts feature.

diest committed:
Correções

Deixei minha feature melhor apropriada para a implementação de seu teste.

Commit:
Update EvaluationComparison.groovy

dansq committed:
Merge pull request #4 from ehammo/master

Merge de testes

cjvg committed:
Merge pull request #4 from gaabs/master

Edited chromedriver file and gebconfig to make tests run.

imvm committed:
Implementacao testes de notificacao

diest committed:
Create GradeCalculator.groovy

Esta classe só tem o método "vazio" calculateGrade para não gerar erro nos testes.

diest committed:
Create FinalGradeController.groovy

Uma interface para o método do qual precisarei nos testes. Ainda preciso incrementar mais para os testes não comprometerem nenhuma funcionalidade.

diest committed:
Create FinalGradeSteps.groovy

Primeira parte da criação dos testes da feature

diest committed:
Update FinalGrade.feature

Mais edições para melhorar a parte dos testes

diest committed:
Mais correções

Corrigi mais imprecisões dos cenários

diest committed:
Update FinalGrade.feature

dansq committed:
Correção de typos/nomes de classes

diest committed:
Delete GradeCalculator.groovy

Não era necessário

dansq committed:
Merge remote-tracking branch 'origin/master'

diest committed:
Interfaces incrementadas

Adicionadas mais interfaces para que o funcionamento não seja prejudicado.

diest committed:
Update FinalGradeSteps.groovy

Não fiz testes de GUI porque não entendi como minha feature deve se comportar nas páginas ou as ações que ela deva praticar em cada passo.

diest committed:
Correção de FinalGradeSteps.groovy

Corrigi alguns erros básicos

renatodms committed:
Merge pull request #1 from dansq/master

Testes groovy

renatodms committed:
Merge branch 'imvm-master'

renatodms committed:
Merge https://github.com/renatodeyvson/TA

Commit:
Merge pull request #6 from cjvg/master

Rearranjo de envio

Commit:
Merge pull request #2 from renatodeyvson/master

3o fase

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

Commit:
Update GebConfig.groovy

Commit:
Update GebConfig.groovy

Commit:
Update GebConfig.groovy

Commit:
Merge pull request #3 from ehammo/master

Merged

Commit:
Merge pull request #4 from diest/master

Commits do aluno 10

Commit:
Corrected many compilation errors.

Tried everything, but couldn't make Tests back to work since merge.

ehammo commented on test/cucumber/steps/FinalGradeSteps.groovy:
```diff
@@ -0,0 +1,63 @@
+//import ta.Student
+//import ta.EvaluationCriterion
+//import ta.FinalGradeController
+//
```
Falta o 
this.metaClass.mixin(cucumber.api.groovy.Hooks)
this.metaClass.mixin(cucumber.api.groovy.EN)

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+Then I can see a detailed table with both student and the professor Evaluations being put, in each criterion, side by side in the screen.
+*/
+
+Given (~'^Given the student "([^"]*)" appear in the list of student that sent their auto-Evaluation$'){
```
Given não deve ser colocado dentro da expressão.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+*/
+
+Given (~'^Given the student "([^"]*)" don`t appear in the list of student that sent their auto-Evaluation$'){
+    String text ->
```
Given não deve ser colocado dentro da expressão.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+
+}
+
+When (~'^choose to compare student "([^"]*)" grades$') {
```
Passos duplicados, igual ao When da linha 24.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+Then I can see a error message with a go-back button to go to the main page.
+*/
+
+Given (~'^Given the student "([^"]*)" don`t appear in the list of student that sent their auto-Evaluation$'){
```
Given não deve ser colocado dentro da expressão.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+Then the system returns a detailed table with both student and the professor grades.
+*/
+
+Given (~'^Given the student "([^"]*)"`s Auto-Evaluation is on the database$'){
```
Given não deve ser colocado dentro da expressão.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+Then the system returns an exception.
+*/
+
+Given (~'^Given the student "([^"]*)"`s Auto-Evaluation isn`t on the database$'){
```
Given não deve ser colocado dentro da expressão.

cjvg commented on test/cucumber/steps/EvaluationComparison.groovy:
```diff
@@ -0,0 +1,115 @@
+import ta.Student
+import ta.StudentController
+
+
```
Falta:
this.metaClass.mixin(cucumber.api.groovy.Hooks)
this.metaClass.mixin(cucumber.api.groovy.EN)

Commit:
Commented whole class AlternativeControllerSpec. It had incompatible commands to our grails version.

Full commented 	accesscontrol and              	autoavaliacaoprazo.

corrected pedro duplicate steps

corrected renato duplicate steps

corrected caio duplicate steps

corrected maia duplicate steps and classes

corrected ian notifications feature and autoavaliacaoprazo

deleted	cadastrodecriterios feature (unknown format)

gaabs commented on test/cucumber/steps/RegisterDiscipline.groovy:
```diff
@@ -0,0 +1,78 @@
+    assert Discipline.findByName(discipline) != null
+}
+
+Then(~'The discipline "([^"]*)" is not stored more than one time in the system$') { String discipline ->
```
Step Given repetido em relação a linha 16 (pode apagar)

dansq committed:
Merge remote-tracking branch 'upstream/master'

imvm committed:
Implementação feature Notifications

dansq committed:
entrega features, correção controlador, views, e domain

Commit:
Merge pull request #7 from gaabs/master

Pegar fork mais atualizado

Commit:
Domain class "Concept" created.

Commit:
Concept views

dansq committed:
arquivo foi modificado na máquina virtual, voltando ao original

jgsjv committed:
Merge pull request #6 from dansq/master

Merge da entrega 21/10/2015

Commit:
Atualizando repositório

imvm committed:
Delete modules.xml

imvm committed:
Setting Travis build

Commit:
Feature de inserção de conceitos manuais nos critérios.

imvm committed:
Deleting configuration file

imvm committed:
Merge https://github.com/imvm/TA

Commit:
Implemented scenario: importing invalid spreadsheet (gui and controller)

Commit:
Merge pull request #6 from psg2/master

Merge 3 >> 4

Commit:
Domain

Commit:
Backup

dansq committed:
Merge pull request #7 from jgsjv/master

Merge pull request #6 from dansq/master

Commit:
merge

dansq committed:
Merge branch 'igormatos-master'

Commit:
commit module

Commit:
Ian Merge

Commit:
Merge branch 'imvm-master'

Commit:
Merge Dansq

Commit:
Merge branch 'dansq-master'

Commit:
Merge branch 'master' of https://github.com/dansq/TA into dansq-master

Conflicts:
	.idea/modules.xml
	grails-app/controllers/ta/ConceptController.groovy
	grails-app/controllers/ta/NotificationController.groovy
	grails-app/controllers/ta/StudentController.groovy
	grails-app/views/student/index.gsp
	test/cucumber/AccessControl.feature
	test/cucumber/AutoavaliacaoPrazo.feature
	test/cucumber/steps/FinalGradeSteps.groovy
	test/functional/pages/ShowNotificationsPage.groovy
	test/functional/steps/NotificationsTestDataAndOperations.groovy

Commit:
Merge branch 'dansq-master'

Commit:
Correção de Conflitos

Commit:
Merge gaabs

Commit:
Merge branch 'gaabs-master'

Commit:
Merge branch 'master' of https://github.com/gaabs/TA into gaabs-master

Conflicts:
	grails-app/controllers/ta/ConceptController.groovy
	grails-app/views/concept/index.gsp
	grails-app/views/concept/upload.gsp

Commit:
Merge branch 'gaabs-master'

Commit:
Correção de Conflitos numero 2

Commit:
Comentários em testes com duplicate steps.

imvm committed:
geb config file

imvm committed:
geb config

imvm committed:
geb config

imvm committed:
pull from fork

imvm committed:
test adjustments

Commit:
Ajustes no Travis, testes e controller'

Commit:
Correção import errado

Commit:
Importação de planilha ok. Falta apenas adicionar os dados importados no estudante, dado que eles está mal estruturado.

Commit:
Correção de testes

Commit:
Alteraçao no caso de arquivo 'csv', pois o plugin nao importa arquivos nesse formato.

Commit:
Concept update, modifications in domain Student and controller Student.
Small updates in Student views as well

Commit:
alteracoes no formato planilha

Commit:
Calculo de media dos criterios

Commit:
Merge branch 'master' of https://github.com/gaabs/TA

Commit:
Merge branch 'Testes' of https://github.com/psg2/TA

Commit:
Alteração na importação de planilhas.
Caso o aluno ou criterio da planilha não exista, ele é cadastrado.

Commit:
Implementacao de Testes

cjvg committed:
Merge pull request #6 from psg2/Testes

Testes

Commit:
Alterações na feature de importação de planilhas
Alteração na importação de planilhas.
Verificação de colunas válidas

Commit:
Adição de form para view e metodo para o submit

Commit:
Adição de cenários e implementação de testes

Commit:
Adição de métodos auxiliares para teste de controller e GUI

Commit:
Correção do caso de erro da GUI

cjvg committed:
Mudanças para feature, steps e student

Commit:
Alterações na feature de importação de planilhas
Alteração na importação de planilhas (conceptController)
Verificação de colunas válidas (sheetImporter)
Alteração nos testes de importação de planilhas

Commit:
Adicionei arquivos auxiliares para testes (planilhas padrão e defeituosas)
Correção e implementação de testes da importação de planilhas

Commit:
Merge branch 'Testes' of https://github.com/psg2/TA

Conflicts:
	test/cucumber/steps/ManualConceptInputSteps.groovy

cjvg committed:
shows auto evaluation

cjvg committed:
changes to update auto evaluation

Commit:
Alterações na importação de planilhas, correção e implementação de respectivos testes

Commit:
Leitura JQuery da página de inserção de conceitos para testes de GUI
Alguns ajutes para o Controller e views de Student

Commit:
Compare Feature

Commit:
Configuração da IDE para Windows
Mudança dos títulos das páginas
Teste GUI semi-completo

Commit:
Adicionei arquivos auxiliares para testes (planilhas padrão e defeituosas)
Correção e implementação de testes da importação de planilhas

Commit:
gebconfig windows

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	test/functional/GebConfig.groovy

Commit:
Compare Feature steps

Commit:
Steps changes

Commit:
ImportConcepts from spreadsheet GUI tests

Commit:
files from gui test (sheet import)

Commit:
Steps and .feature changes
Pages and views updates

Commit:
ImportConcepts from spreadsheet  tests

Commit:
coloring differences

Commit:
-unit

Commit:
Finalização dos 8 cenários de teste.

cjvg committed:
changed methods for auto evaluation and the page

Commit:
Correção e implementação de testes da importação de planilhas

Commit:
[completed #107523732] [completed #109086474] Finalização dos 8 testes da Feature Manual Concept Input

Commit:
Tentativa de corrigir falhas nos testes. Há um problema de alteração concorrente no DB ainda não resolvido, e, aparentemente, o Travis não consegue compilar o Excel Importer.

Commit:
travis verbose

Commit:
test updates

imvm committed:
updating branch

Commit:
Adição de css para página da feature

Commit:
Steps Comentado

Commit:
Merge pull request #7 from ehammo/Testes

Steps Comentado

Commit:
Merge pull request #14 from psg2/Testes

Merge com branch

Commit:
Para o travis

Commit:
Para o travis - commit 2

cjvg committed:
[109091068] controller test for auto evaluation

dansq committed:
adicionando novos cenários e testes para RegisterDiscipline

renatodms committed:
Cenários de atualização de dados de professores

renatodms committed:
Merge pull request #3 from psg2/master

atualizar

imvm committed:
travis fix

imvm committed:
merge renato

imvm committed:
removed duplicated steps

imvm committed:
geb fix

dansq committed:
merge

dansq committed:
merge

cjvg committed:
Merge branch 'master' of https://github.com/gaabs/TA into gaabs-master

# Conflicts:
#	grails-app/controllers/ta/StudentController.groovy

cjvg committed:
Merge branch 'gaabs-master'

dansq committed:
erge

dansq committed:
Merge branch 'ehammo-master'

imvm committed:
Adjusting Notification controller and domain classes

imvm committed:
Fixed incorrect name error

imvm committed:
Removed duplicated steps

imvm committed:
Organized steps file

imvm committed:
Removed duplicate step, organized steps file

imvm committed:
Removed unnecessary link

imvm committed:
[#107883384]Developing scenario test

imvm committed:
[#107883384]Updated feature file

imvm committed:
[#107883384]Updated Notification class attributes

imvm committed:
[#107883384]Updated feature file

imvm committed:
[#107883384]Developing scenario tests

imvm committed:
[#107883412]Developing scenario tests

imvm committed:
[#107883384]Developing scenario tests

imvm committed:
[#107883384][#107883412]Updating steps

imvm committed:
[#107883384][#107883412]Changed test student name and login

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
Added ignore

imvm committed:
[#107883384][#107883412]Developing tests

imvm committed:
Reverted local testing change

imvm committed:
Fixed ChromeDriver import

imvm committed:
Fixing Travis invalid variable bug

imvm committed:
Fixing comment

imvm committed:
Removed comment

Commit:
merge cjvg

Commit:
Merge branch 'cjvg-master'

Commit:
[#107523050] + Merge imvm

Commit:
Merge branch 'imvm-master'

Commit:
Update ImportConcepts.feature

Ignorando cenários por conta da incompatibilidade do Travis com o plug-in do Excel utilizado

Commit:
para o travis - 1 merge

Commit:
Merge pull request #8 from dansq/master

2a fase da entrega

Commit:
para o travis - 2 merge

Commit:
para o travis - 2 merge - corrigindo erros de compilacao do RegisterDiscipline.groovy

Commit:
para o travis - 2 merge - corrigindo erros de compilacao do RegisterDiscipline.groovy

renatodms committed:
Merge pull request #4 from gaabs/master

Entrega 2a iteração: 1 + 5 >> 7

Commit:
@ignore nos meus testes e modificações para compatibilizar com testes de contribuidores

Commit:
Merge branch 'master' of https://github.com/gaabs/TA

Conflicts:
	test/cucumber/ImportConcepts.feature

Commit:
Tentativa de compatibilizar testes da equipe

Commit:
Tentativa de compatibilizar testes da equipe

Commit:
Resolução de conflitos

Commit:
Resolução para merge

Commit:
Merge branch 'ehammo-master'

Commit:
Correção de conflitos remanescentes do merge.
Correção do GebConfig para rodar no Travis
Adição do método deleteAfterTest()

renatodms committed:
Merge pull request #5 from gaabs/master

Pull request com alterações

Commit:
Correção do método deleteAfterTest

Commit:
Correção de conflitos para segundo merge

Commit:
Merge branch 'renatodeyvson-master'

Commit:
Correção de conflitos pós merge.

jpms2 committed:
Merge pull request #1 from gaabs/master

tentativa de passar teste do excel

jpms2 committed:
Update ImportConcepts.feature

jpms2 committed:
Update BuildConfig.groovy

Commit:
Merge pull request #9 from jpms2/master

Correção travis-planilha

Commit:
Merge https://github.com/gaabs/TA

Commit:
Teste travis

Commit:
Remoção do CSS
Remoção dos critérios removidos das hash contidas nos alunos
Criação da classe getPageTitle para os testes de GUI serem compatíveis com qualquer língua disponível
Refatoração de código duplicado em Student na inicialização das hashs.
Refatoração de código duplicado em StudentPage para verificação de conceitos e critérios nos testes de GUI.
Correção de um teste antigo com erro de digitação

Commit:
[#110668934] Refatoração da importação de conceitos (planilha)
- Correção dos testes de planilha com Travis
- Adição da limpeza do BD antes de cada teste
- Refatoração de SheetController, SheetImporter, ImportConcepts.feature, ImportConcepts.groovy.
- Adição da classe auxiliar para testes de GUI GetPageTitle
- Adição da planilha para testes ValidSheet2

Commit:
[#111367600] Adição de novos cenários de GUI
Ajuste de cenários antigos para melhorar a complexidade
Ajuste nas pages para funcionar com Locale e GetPageTitle
Correção da configuração de limpar o banco de dados após cada teste para funcionar com GUI

Commit:
fixed untracked files

Commit:
Merge https://github.com/pauloborba/TA

Conflicts:
	test/functional/GebConfig.groovy

Commit:
Merge https://github.com/pauloborba/TA

Conflicts:
	test/functional/GebConfig.groovy

Pages corrections

Commit:
[#110668934] Refatoração da SheetUploadPage

Commit:
[#110668934] Refatoração

Commit:
[completed #110668934] Finalização da Refatoração da importação de conceitos
- Remoção de método inutilizado click() na SheetUploadPage
- Adição do diretório ".slcache" ao gitignore

Commit:
gebConfig

Commit:
Merge master

Commit:
Merge pull request #17 from pauloborba/master

Merge master

Commit:
[ Completed #111367600 #111367594 ] Adição de novos cenários, correção de antigos, refatoração e ajustes nas Pages.

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Travis Update

Commit:
Merge https://github.com/pauloborba/TA
Travis fix

# Conflicts:
#	.travis.yml

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

# Conflicts:
#	.idea/modules.xml
#	test/cucumber/EvaluateStudent.feature
#	test/cucumber/ManualConceptInput.feature
#	test/cucumber/steps/ManualConceptInputSteps.groovy
#	test/cucumber/support/env.groovy
#	test/functional/GebConfig.groovy
#	test/functional/pages/EvaluationCriterionPages/CreateEvaluationCriterionPage.groovy
#	test/functional/pages/EvaluationCriterionPages/EvaluationCriterionPage.groovy
#	test/functional/pages/GetPageTitle.groovy
#	test/functional/pages/ManualInputPage.groovy
#	test/functional/pages/RegisterEvaluationPage.groovy
#	test/functional/pages/ShowEvaluationPage.groovy
#	test/functional/pages/StudentPages/CreateStudentPage.groovy
#	test/functional/pages/StudentPages/StudentPage.groovy

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

# Conflicts:
#	.idea/modules.xml
#	test/cucumber/EvaluateStudent.feature
#	test/cucumber/ManualConceptInput.feature
#	test/cucumber/steps/ManualConceptInputSteps.groovy
#	test/cucumber/support/env.groovy
#	test/functional/GebConfig.groovy
#	test/functional/pages/EvaluationCriterionPages/CreateEvaluationCriterionPage.groovy
#	test/functional/pages/EvaluationCriterionPages/EvaluationCriterionPage.groovy
#	test/functional/pages/GetPageTitle.groovy
#	test/functional/pages/ManualInputPage.groovy
#	test/functional/pages/RegisterEvaluationPage.groovy
#	test/functional/pages/ShowEvaluationPage.groovy
#	test/functional/pages/StudentPages/CreateStudentPage.groovy
#	test/functional/pages/StudentPages/StudentPage.groovy

Commit:
Merge branch 'psg2-master'
